dialogue-okhttp-client/src/main/java/com/palantir/dialogue/OkHttpChannel.java

```diff
@@ -93,6 +93,11 @@ public void writeTo(BufferedSink sink) throws IOException {
             case PATCH:
                 okRequest = okRequest.patch(toOkHttpBody(request.body()));
                 break;
+            case OPTIONS:
+                Preconditions.checkArgument(
+                        !request.body().isPresent(), "OPTIONS endpoints must not have a request body");
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/OPTIONS
>Request has body: No

Reply:
We have tests for failing GET/HEAD requests with bodies, mind adding a variant to test OPTIONS with a request body?